openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -310,6 +332,30 @@ def publish_from_drafts(
         published_at = datetime.now(tz=timezone.utc)
 
     with atomic():
+        # If the drafts include any containers, we need to auto-publish their descendants:
+        # TODO: this only handles one level deep and would need to be updated to support sections > subsections > units
+
+        # Get the IDs of the ContainerVersion for any Containers whose drafts are slated to be published.
+        container_version_ids = (
+            Container.objects.filter(publishable_entity__draft__in=draft_qset)
+            .values_list("publishable_entity__draft__version__containerversion__pk", flat=True)
+        )
+        if container_version_ids:
+            # We are publishing at least one container. Check if it has any child components that aren't already slated
+            # to be published.
+            unpublished_draft_children = EntityListRow.objects.filter(
+                entity_list__container_versions__pk__in=container_version_ids,
+                entity_version=None,  # Unpinned entities only
```

Comment:
I have this weird recollection that it was necessary to do something like `entity_version__isnull=True` rather than `entity_version=None` here, but that doesn't seem to be the case based on some tinkering I just did on the console. I don't know if I'm just completely misremembering, or if Django behavior has changed, or else if there was some weird edge case. No change needed here, but I'm writing this in case it triggers any bells for you.

Reply:
I have a similar vague recollection but I think maybe we're just thinking of how you have to write the queries when using raw SQL, where `= NULL` is always false but `IS NULL` gives the correct result. According to [this](https://stackoverflow.com/questions/29858701/django-query-isnull-true-or-none) they're mostly the same in Django except for some one particular case involving PostgreSQL JSON field comparisons.

Reply:
Oh! Now I remember. This is what I was thinking about:

https://github.com/openedx/openedx-learning/blob/4316b5cb45faa974a058716d2d5edb0658b7dd24/openedx_learning/apps/authoring/publishing/api.py#L237-L247

(So not a problem with this piece of code.)